### PR TITLE
T6792: Autocomplete for "show arp interface" interfaces

### PR DESCRIPTION
**Pull request by HollyGurza**
extended filter for possible broadcast-capable interfaces

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6792

## How to test

add an interface with allowed type e.g dummy:
```
conf
set interface dummy dum0 address 192.168.1.1/24
commit
```
see possible completion:
```
# run show arp interface 
Possible completions:
  dum0                  Show Address Resolution Protocol (ARP) cache for specified
                        interface
  eth0
  eth1
  eth2
  eth3
```
add and check ARP records
```
# sudo ip neigh add 192.168.1.2 lladdr 00:11:22:33:44:55 dev dum0
# run show arp interface dum0 
Address      Interface    Link layer address    State
-----------  -----------  --------------------  ---------
192.168.1.2  dum0         00:11:22:33:44:55     PERMANENT
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/completion/list_interfaces/list_interfaces.ml [hunk 1]
```diff
@@ -68,7 +68,7 @@ let filter_from_type it =
     | _ -> Some (filter_from_prefix pre)
 
 let filter_broadcast s =
-    let pattern = {|^(bond|br|eth)(.*)$|}
+    let pattern = {|^(bond|br|dum|eth|gnv|peth|macsec|vxlan|wwan|wlan)(.*)$|}
```
Comment (sever-sever) [inline src/completion/list_interfaces/list_interfaces.ml]: Are you sure the `dum` interface (and others) supports ARP?
```
vyos@r14:~$ show int
Codes: S - State, L - Link, u - Up, D - Down, A - Admin Down
Interface    IP Address         MAC                VRF        MTU  S/L    Description
-----------  -----------------  -----------------  -------  -----  -----  -------------
dum0         203.0.113.1/24     be:62:fd:ba:50:50  default   1500  u/u
eth0         192.168.122.14/24  52:54:00:77:fa:36  default   1500  u/u
lo           127.0.0.1/8        00:00:00:00:00:00  default  65536  u/u
             ::1/128
veth0        -                  0a:d6:2d:f8:bd:e3  default   1500  u/u
vyos@r14:~$ 
vyos@r14:~$ 
vyos@r14:~$ ping 203.0.113.1
PING 203.0.113.1 (203.0.113.1) 56(84) bytes of data.
64 bytes from 203.0.113.1: icmp_seq=1 ttl=64 time=0.034 ms
64 bytes from 203.0.113.1: icmp_seq=2 ttl=64 time=0.041 ms
^C
--- 203.0.113.1 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1018ms
rtt min/avg/max/mdev = 0.034/0.037/0.041/0.003 ms
vyos@r14:~$ 
vyos@r14:~$ 
vyos@r14:~$ 
vyos@r14:~$ show arp interface dum0
Address    Interface    Link layer address    State
---------  -----------  --------------------  -------
vyos@r14:~$ 
vyos@r14:~$ 
vyos@r14:~$ show arp
Address         Interface    Link layer address    State
--------------  -----------  --------------------  ---------
192.168.122.15  eth0         52:54:00:55:80:8c     DELAY
192.168.122.1   eth0         52:54:00:16:84:08     REACHABLE
vyos@r14:~$ 

```

Needs to check other interfaces before adding

**Review by sever-sever — DISMISSED** [review]

File: src/completion/list_interfaces/list_interfaces.ml (hunk 1 above)
Comment (dmbaturin) [inline src/completion/list_interfaces/list_interfaces.ml]:
```suggestion
    let pattern = {|^(bond|br|tun|vtun|eth|gnv|peth|macsec|vxlan|wwan|wlan)(.*)$|}
```

Comment (dmbaturin) [reply src/completion/list_interfaces/list_interfaces.ml]: Dummy interfaces don't use ARP, but tunnels in the `gre-bridge` mode and OpenVPN do. Let's add those.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: src/completion/list_interfaces/list_interfaces.ml (hunk 1 above)
Comment (HollyGurza) [reply src/completion/list_interfaces/list_interfaces.ml]: dum removed from allow list

**Review by dmbaturin — APPROVED** [review]

**Review by sever-sever — APPROVED** [review]

**Pull request merged**